src/main/java/Duke.java

```diff
@@ -1,10 +1,228 @@
+    private static final String PARAM_BY = "/by";
+
+    /**
+     * Handles user I/O.
+     */
+    private static final Scanner SC = new Scanner(System.in);
+
+    /**
+     * Stores a list of user tasks
+     */
+    private static final ArrayList<Task> TASKS = new ArrayList<Task>();
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        printWelcome();
+        while (SC.hasNextLine()) {
+            String input = getUserInput();
+            executeUserCommand(input);
+        }
+    }
+
+    /**
+     * Displays welcome message.
+     */
+    public static void printWelcome() {
+        System.out.println(MESSAGE_INTRO);
+        System.out.println(MESSAGE_BODY);
+    }
+
+    /**
+     * Displays the completed task.
+     * @param taskNo user specified task number
+     */
+    public static void printDone(int taskNo) {
+        printWithDividers(MESSAGE_DONE_TASK, TASKS.get(taskNo).toString());
+    }
+
+    /**
+     * Displays a list of tasks available.
+     */
+    public static void printListOfTasks() {
+        System.out.println(DIVIDER);
+        System.out.println(MESSAGE_LIST_TASK);
+        for (int i = 0; i < TASKS.size(); i++) {
+            System.out.println((i+1) + ". " + TASKS.get(i));
+        }
+        System.out.println(DIVIDER);
+    }
+
+    /**
+     * Displays message whenever a new task is added.
+     * @param task user specified Task
+     */
+    public static void printAddTask(Task task) {
+        String taskCount = "Now you have " + TASKS.size() + " tasks in the list.";
+        printWithDividers(MESSAGE_ADD_TASK, task.toString(), taskCount);
+    }
+
+    /**
+     * Displays a message enclosed within dividers.
+     * @param message user specified message(s)
+     */
+    public static void printWithDividers(String... message) {
+        System.out.println(DIVIDER);
+        for (String m : message) {
+            System.out.println(m);
+        }
+        System.out.println(DIVIDER);
+    }
+
+    /**
+     * Displays goodbye message and exits the program.
+     */
+    public static void exitProgram() {
+        System.out.println(MESSAGE_GOODBYE);
+        System.exit(0);
+    }
+
+    /**
+     * Reads texts entered by user on command line
+     * @return a String of text entered by the user
+     */
+    public static String getUserInput() {
+        return SC.nextLine().trim();
+    }
+
+    /**
+     * Handles and executes command entered by user
+     * @param input raw input from user
+     */
+    public static void executeUserCommand(String input) {
+        String[] processedInputs = processUserInput(input);
+        String command = processedInputs[0];
+        String args = processedInputs[1];
+
```

Comment:
Looks good! I really like the use of constants in switch statements. It makes things clear